.github/workflows/merge.yml

```
@@ -0,0 +1,55 @@
+name: Merge
+
+on:
+  push:
+    branches: [ "main" ]
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Just to make it clearer what this job does it might make sense to name them accordingly, e.g.,:
```suggestion
  buildandunittest:
    runs-on: ubuntu-latest
    name: Build and Run Unit Tests
```